benchmark/src/androidTest/java/com/apollographql/apollo/benchmark/Benchmark.kt

```diff
@@ -72,6 +71,8 @@ class Benchmark {
     val data = operation.parse(bufferedSource()).data!!
 
     val records = operation.normalize(data, CustomScalarAdapters.DEFAULT, CacheKeyResolver.DEFAULT)
+
+    val cache = apolloClient.apolloStore.normalizedCache()
```

Comment:
(not directly related to this PR) Would it make sense in the near future to have some MemoryCache benchmarks as well? Just to catch any regressions/improvements 😃

Reply:
Yep, definitely, see https://github.com/apollographql/apollo-android/pull/2909

I was more expecting multi-threading/contention to be the main issue on memory caches but you're right, maybe the batching introduces a cost there (for an exemple, it could allocate slightly more for all the pending references). I'll add that.

Reply:
```
benchmark:     4,657,449 ns Benchmark.apolloReadCacheMemory
benchmark:     7,913,594 ns Benchmark.apolloBatchCacheMemory
benchmark:   232,572,731 ns Benchmark.apolloReadCacheSql
benchmark:    69,434,069 ns Benchmark.apolloBatchCacheSql
```

Batching slows down memory reading by almost a factor ~x2 and speeds up SQL by ~x3...

Reply:
Yeah good point 👍 

In our Falcor batching implementation, we definitely saw a small increase in memory used, but it was alleviated by the speed bump from SQL queries batching. Not sure about Memory Batching (even if we do it, it's invisible to the UIs and I don't think we explicitly measured it 🤔)

Reply:
I would say the raw loss in MemoryCache (4ms) is small compared to the big raw gain in SQL (160ms)?

Interestingly, in the field I've seen the `LRUMemoryCache`(relying on nytimes dependencies) response time more in the order of 100ms, do you think there were incremental improvements with the new `MemoryCache`? Or the 100ms vs 8ms could just be from multi-threading and contention like you mentioned.

Reply:
Oh wow, ~100ms to read from a memory cache sound a lot... If you ever get a chance to log a systrace, that'd be interesting to look at. 

I think it make sense to not batch if it's only memory and always batch if it's only SQL. Now if there is memory + SQL then I'm not sure. I'd say the 160ms are more important indeed but maybe it depends the use cases...

Reply:
To be fair this 100ms is the time between a query is launched and when ApolloClient answers with Operation.Data (with thread switching etc...), so it's pretty good!